### ISSUE TITLE
Error/lint on non-use of return values once purity is completely tracked

### ISSUE THREAD

**Issue by calebzulawski** [CACI-International/ergo#20]
I just spent some time debugging a simple case where I meant to write `^func args` and instead had `func args`

**Comment by afranchuk** [comment]
Yeah this would be very easy if the function is known to be pure. Unfortunately if the function is impure the same mistake can be made :( If there were more type information (static analysis) available, there could be a lint that warns when e.g. a function returns a map but isn't being merged.

**Comment by calebzulawski** [comment]
Good point, checking the type would probably be better.  This seems like a relatively lightweight check, some lints like this could probably be turned on by default (without specifically enabling linting)
